app/src/main/java/com/es/findsoccerplayers/LoginActivity.java

```diff
@@ -0,0 +1,193 @@
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
+            }
+        });
+
+        btnGoogle.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signIn();
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                GoogleSignInAccount account = task.getResult(ApiException.class);
+                firebaseAuthWithGoogle(account.getIdToken());
+                Toast.makeText(LoginActivity.this, "Loggato con successo", Toast.LENGTH_SHORT).show();
+            } catch (ApiException e) {
+                // Google Sign In failed, update UI appropriately
+                Toast.makeText(this, e.toString(), Toast.LENGTH_SHORT).show();
+                prgBar.setVisibility(View.INVISIBLE);
+                // ...
+            }
+        }
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        prgBar.setVisibility(View.INVISIBLE);
+        Log.w("LoginAct","Act Creata");
+    }
+
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
+        mAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (task.isSuccessful()) {
+                            // Sign in success, update UI with the signed-in user's information
+                            FirebaseUser user = mAuth.getCurrentUser();
+                            updateUI(user);
+                        } else {
+                            // If sign in fails, display a message to the user.
+                            prgBar.setVisibility(View.INVISIBLE);
+                            Toast.makeText(LoginActivity.this, (CharSequence) task.getException(), Toast.LENGTH_SHORT).show();
+                            updateUI(null);
+                        }
+                    }
+                });
+    }
+
+
+    private void signIn() {
+        prgBar.setVisibility(View.VISIBLE);
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    private void updateUI(FirebaseUser user) {
```

Comment:
`user` a che serve?

Reply:
ho semplicemente copiato e incollato quello che c'era sul sito di android develop, in questo caso non serve

Reply:
magari togli questo metodo, perché come ho scritto nei commenti sotto, la main activity parte solo se il login è andato a buon fine. Se è questo il caso fai direttamente lo start della main activity (visto updateUI fa solo quello)